backend/src/test/java/org/example/backend/controller/MovieControllerTest.java

```diff
@@ -36,18 +34,66 @@ class MovieControllerTest {
     @Autowired
     private ObjectMapper objectMapper;
 
+    @Autowired
+    private MovieRepository movieRepository;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        movieRepository.deleteAll();
+    }
+
     @Test
     void getAllMovies_shouldReturnEmptyList_WhenCalledInitially() throws Exception {
-        mockMvc.perform(MockMvcRequestBuilders.get(BASE_URL)
+        mockMvc.perform(get(BASE_URL)
                         .contentType("application/json"))
-                .andExpect(MockMvcResultMatchers.status().isOk())
-                .andExpect(MockMvcResultMatchers.content().json("[]"));
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(0)));
+    }
+
+    private void addTestMovies() throws Exception {
+        Movie movie1 = new Movie();
+        movie1.setImdbId("tt1234567");
+        movie1.setTitle("Test Movie");
+        movie1.setReleaseDate("2021-01-01");
+        movie1.setTrailerLink("https://www.youtube.com/watch?v=1234567");
+        movie1.setPoster("https://www.imdb.com/title/tt1234567/");
+        movie1.setGenres(List.of("Action", "Adventure"));
+        movie1.setBackdrops(List.of("https://www.imdb.com/title/tt1234567/mediaviewer/rm1234567"));
+
+        Movie movie2 = new Movie();
+        movie2.setImdbId("tt7654321");
+        movie2.setTitle("Test Movie 2");
+        movie2.setReleaseDate("2021-01-01");
+        movie2.setTrailerLink("https://www.youtube.com/watch?v=1234567");
+        movie2.setPoster("https://www.imdb.com/title/tt1234567/");
+        movie2.setGenres(List.of("Action", "Adventure"));
+        movie2.setBackdrops(List.of("https://www.imdb.com/title/tt1234567/mediaviewer/rm1234567"));
+
+        mockMvc.perform(post(BASE_URL + "/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(movie1)))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(post(BASE_URL + "/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(movie2)))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void getAllMovies_ShouldReturnAllMovieList_WhenMoviesAreAdded() throws Exception {
+        addTestMovies();
+
+        mockMvc.perform(get(BASE_URL)
+                        .contentType("application/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)));
     }
 
     @Test
     void getSingleMovie_ShouldReturnMovie_WhenCalledWithValidImdbId() throws Exception {
 
-        mockMvc.perform(post("/api/v1/movies/add")
+        mockMvc.perform(post(BASE_URL + "/add")
```

Comment:
Sehr gut, denk bitte nur daran:
Die Post Methode ist nur das Setup, damit du die GetById machen kannst.

Du musst der Post Methode also noch folgendes geben:
- ein .andReturn() um den Wert in einer Variablen abzuspeichern
- Diese Variable wandelst du dann mit dem Objectmapper in ein Movie Object um
- Hiervon ließt du die ID aus, damit du diese in die URL deiner getByID Methode einfügen kannst